Mesh_3/include/CGAL/Mesh_3/Protect_edges_sizing_field.h

```diff
@@ -1490,9 +1522,42 @@ do_balls_intersect(const Vertex_handle& va, const Vertex_handle& vb) const
   return do_intersect(sphere(cp(wa), cw(wa)), sphere(cp(wb), cw(wb)));
 }
 
-template <typename C3T3, typename MD, typename Sf>
-typename Protect_edges_sizing_field<C3T3, MD, Sf>::Vertex_handle
-Protect_edges_sizing_field<C3T3, MD, Sf>::
+template <typename C3T3, typename MD, typename Sf, typename Df>
+bool
+    Protect_edges_sizing_field<C3T3, MD, Sf, Df>::
+    approx_is_too_large(const Vertex_handle& va, const Vertex_handle& vb) const
+{
+  if ( ! c3t3_.is_in_complex(va, vb) )
+  {
+    return false;
+  }
+  typedef typename Kernel::Point_3 Point_3;
+
+  Curve_index curve_index = domain_.curve_index((va->in_dimension() < vb->in_dimension()) ? vb->index() : va->index());
+
+  Point_3 pa = va->point().point();
+  Point_3 pb = vb->point().point();
+  Point_3 segment_middle = CGAL::midpoint(pa, pb);
+  // Obtain the geodesic middle point
+  FT signed_geodesic_distance = domain_.signed_geodesic_distance(pa, pb, curve_index);
+  Point_3 geodesic_middle;
+  if (signed_geodesic_distance >= FT(0))
+  {
+    geodesic_middle = domain_.construct_point_on_curve(pa, curve_index, signed_geodesic_distance / 2);
+  }
+  else
+  {
+    geodesic_middle = domain_.construct_point_on_curve(pb, curve_index, -signed_geodesic_distance / 2);
+  }
+  // Compare distance to the parameter's distance
+  FT squared_evaluated_distance = CGAL::squared_distance(segment_middle, geodesic_middle);
+  FT segment_middle_edge_distance = query_distance(segment_middle, 1, curve_index);
```

Comment:
above you mentioned "squared distance". From here I guess it's not. Can you please make sure it's consistent?